pkg/authn/keychain.go

```diff
@@ -24,13 +24,17 @@ import (
 	"runtime"
 
 	"github.com/google/go-containerregistry/pkg/logs"
-	"github.com/google/go-containerregistry/pkg/name"
 )
 
+type Target interface {
```

Comment:
Target isn't a great name, document this and members?

Reply:
Better idea than `Target`? `Principal`? `Authority`?

Reply:
`authn.Target` seems fine when qualified. I'm open to better suggestions.